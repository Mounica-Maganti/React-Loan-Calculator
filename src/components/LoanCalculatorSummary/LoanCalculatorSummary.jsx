import React, { Fragment } from 'react';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import PropTypes from 'prop-types';
import * as _ from 'lodash';
import { LoanSummaryLocale } from '../../locale/index';
import LoanSummaryDetails from '../LoanSummaryDetails/LoanSummaryDetails';


const StyledTableCell = withStyles((theme) => ({
  head: {
    backgroundColor: theme.palette.common.black,
    color: theme.palette.common.white,
  },
  body: {
    fontSize: 14,
  },
}))(TableCell);

const StyledTableRow = withStyles((theme) => ({
  root: {
    '&:nth-of-type(odd)': {
      backgroundColor: theme.palette.action.hover,
    },
  },
}))(TableRow);

const useStyles = makeStyles({
  table: {
    minWidth: 700,
  },
});

const LoanCalculatorSummary = ({ loanInfo, summaryData }) => {
  const classes = useStyles();

  return (
    <Fragment>
      <LoanSummaryDetails {...summaryData} />
      <TableContainer component={Paper} className="loan-summary-container">
        <Table className={classes.table} aria-label="customized table">
          <TableHead>
            <TableRow>
              {
                _.get(LoanSummaryLocale, 'tableColumns', []).map((column, index) => (
                  <StyledTableCell
                    key={index}
                    align={"center"}
                    id={`${column}ColumnHeader`}
                  > {column}</StyledTableCell>
                ))
              }
            </TableRow>
          </TableHead>
          <TableBody>
            {loanInfo.map((row) => (
              <StyledTableRow key={row.monthCount} id="tableBody">
                <StyledTableCell align="center" component="th" scope="row">
                  {row.monthCount}
                </StyledTableCell>
                <StyledTableCell align="center">{row.balance}</StyledTableCell>
                <StyledTableCell align="center">{row.interest}</StyledTableCell>
                <StyledTableCell align="center">{row.principal}</StyledTableCell>
              </StyledTableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </Fragment>
  );
}

LoanCalculatorSummary.propTypes = {
  loanInfo: PropTypes.array.isRequired,
  summaryData: PropTypes.object.isRequired,
};

LoanCalculatorSummary.defaultProps = {
  loanInfo: [],
  summaryData: {}
};

export default LoanCalculatorSummary;