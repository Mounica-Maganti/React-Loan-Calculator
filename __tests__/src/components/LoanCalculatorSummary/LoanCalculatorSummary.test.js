import React from 'react';
import { shallow } from 'enzyme';
import LoanCalculatorSummary from '../../../../src/components/LoanCalculatorSummary/LoanCalculatorSummary';


describe('Loan Calculator Summary Tests', () => {
    let props;
    beforeEach(() => {
        props = {
            loanInfo: [
                {
                    monthCount: 1,
                    balance: 123,
                    interest: 2,
                    principle: 2354
                }
            ],
            summaryData: {}
        }
    });

    it('should render Loan Calculator Component', () => {
        const renderedModule = shallow(<LoanCalculatorSummary {...props} />);
        expect(renderedModule).toMatchSnapshot();
        expect(renderedModule.find('#MonthColumnHeader').length).toBe(1);
        expect(renderedModule.find('#BalanceColumnHeader').length).toBe(1);
        expect(renderedModule.find('#InterestColumnHeader').length).toBe(1);
        expect(renderedModule.find('#PrincipalColumnHeader').length).toBe(1);
        expect(renderedModule.find('.loan-summary-container').length).toBe(1);
        expect(renderedModule.find('#tableBody').length).toBe(1);
    });

});