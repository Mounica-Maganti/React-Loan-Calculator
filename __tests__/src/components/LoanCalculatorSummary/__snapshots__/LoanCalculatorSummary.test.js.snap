// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loan Calculator Summary Tests should render Loan Calculator Component 1`] = `ShallowWrapper {}`;
